src/osd/scrubber/osd_scrub_sched.cc

```diff
@@ -800,3 +800,22 @@ void ScrubQueue::dump_scrub_reservations(ceph::Formatter* f) const
   f->dump_int("scrubs_remote", scrubs_remote);
   f->dump_int("osd_max_scrubs", conf()->osd_max_scrubs);
 }
+
+void ScrubQueue::clear_pg_scrub_blocked(spg_t blocked_pg)
+{
+  dout(5) << fmt::format(": pg {} is unblocked", blocked_pg) << dendl;
+  --blocked_scrubs_cnt;
+  ceph_assert(blocked_scrubs_cnt >= 0);
+}
+
+void ScrubQueue::mark_pg_scrub_blocked(spg_t blocked_pg)
+{
+  dout(5) << fmt::format(": pg {} is blocked on an object", blocked_pg)
+	  << dendl;
+  ++blocked_scrubs_cnt;
```

Comment:
IIUC, we increment `blocked_scrubs_cnt` and later on use this counter (`get_blocked_pgs_count()`) as the number of PGs blocked.
We might have a single blocking PG while reporting that few PGs are blocked.
The first comment should resolve this. (with a per PG count)

Reply:
Resolved offline.
Each PG is guaranteed to have a single instance of a `blocked_range_t` object.